frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -51,10 +51,7 @@ export const PublicFormProvider = ({
 
   const expiryInMs = useMemo(() => {
     if (!vfnTransaction?.expireAt) return null
-    return differenceInMilliseconds(
-      new Date(vfnTransaction.expireAt),
-      Date.now(),
-    )
+    return differenceInMilliseconds(vfnTransaction.expireAt, Date.now())
```

Comment:
It cannot be negative unless our server screws up (or has a MASSIVE 4 hour time drift)

Reply:
Hmm, or the client clock is messed up. I suppose I was asking whether `expireAt` is computed server side or client side 🤔 .  If computed server side, we're comparing clock values of different systems, with one (the client) that we shouldn't trust. If on the other hand `expireAt` is computed client-side as `Date.now() + expiryOffsetInMs`, then the comparison would be safer.

Reply:
GRRRRRRRRRRRR. Dates man... Yea, user clock may screw it up. We currently return `expireAt` as a Date.

Reply:
I think we should not deal with that case though, when they attempt to submit the server will still reject the signature (since the token has expired). This was just a QOL improvement for clients with relatively accurate clocks.

Reply:
Roger that, and agreed 👍  No need to change this (at least for now)!

I've had so much trouble with dates before that I can just "smell" possible problems now 😅